desktop/core/src/desktop/js/apps/storageBrowser/StorageFilePage/StorageFilePage.tsx

```diff
@@ -29,29 +38,50 @@ import {
   SupportedFileTypes
 } from '../../../utils/constants/storageBrowser';
 import { Spin } from 'antd';
+import useLoadData from '../../../utils/hooks/useLoadData';
 
-const StorageFilePage = ({ fileData }: { fileData: PathAndFileData }): JSX.Element => {
+interface StorageFilePageProps {
+  fileName: string;
+  fileStats: FileStats;
+}
+
+const StorageFilePage = ({ fileName, fileStats }: StorageFilePageProps): JSX.Element => {
   const { t } = i18nReact.useTranslation();
-  const [isEditing, setIsEditing] = React.useState(false);
-  const [fileContent, setFileContent] = React.useState(fileData.view?.contents);
-  const fileMetaData = useMemo(() => getFileMetaData(t, fileData), [t, fileData]);
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const [fileContent, setFileContent] = useState<FilePreview['contents']>();
 
   const { loading: isSaving, save } = useSaveData(SAVE_FILE_API_URL);
 
+  const { data: fileData, loading: loadingPreview } = useLoadData<FilePreview>(
```

Comment:
Now I'm a bit confused by this:
We get back a fileData object (that I guess still is kept as state in useLoadData). The fileData object contains "contents". But we also set the fileData.contents in the onSuccess in a fileContent state of this component.. so now that state lives in two places. Is that needed, if so why?

Reply:
Great question.
fileData is the object which will contain the latest record from the backend.
Only the copy of fileData.content is being saved in localstate `fileContent` so that the textarea can update the value and we can check if fileContent is edited or not `fileContent === fileData?.contents`